drivers/include/lis2dh12.h

```diff
@@ -118,89 +180,62 @@ enum {
     LIS2DH12_OK    =  0,            /**< everything was fine */
     LIS2DH12_NOBUS = -1,            /**< bus interface error */
     LIS2DH12_NODEV = -2,            /**< unable to talk to device */
-    LIS2DH12_NOINT = -3,            /**< wrong interrupt line (has to be LIS2DH12_INT1 or LIS2DH12_INT2) */
+    LIS2DH12_NOINT = -3,            /**< wrong interrupt line (has to be LIS2DH12_INT1
+                                        or LIS2DH12_INT2) */
     LIS2DH12_NODATA= -4,            /**< no data available */
 };
 
 #if MODULE_LIS2DH12_INT || DOXYGEN
 /*
  * @brief Interrupt lines
  */
-enum{
+enum {
     LIS2DH12_INT1 = 1,              /**< first interrupt line */
     LIS2DH12_INT2 = 2,              /**< second interrupt line */
 };
 
-/**
- * @brief   Interrupt config register values
- */
-enum {
-    LIS2DH12_INT_CFG_XLIE = 0x01,   /**< enable X low evnt */
-    LIS2DH12_INT_CFG_XHIE = 0x02,   /**< enable X high event */
-    LIS2DH12_INT_CFG_YLIE = 0x04,   /**< enable Y low event */
-    LIS2DH12_INT_CFG_YHIE = 0x08,   /**< enable Y high event */
-    LIS2DH12_INT_CFG_ZLIE = 0x10,   /**< enable Z low event */
-    LIS2DH12_INT_CFG_ZHIE = 0x20,   /**< enable Z high event */
-    LIS2DH12_INT_CFG_6D   = 0x40,   /**< enable 6-direction detection */
-    LIS2DH12_INT_CFG_AOI  = 0x80,   /**< and/or combination interrupt events */
-};
-
-/**
- * @brief   Interrupt type values
- */
-enum {
-    /* for interrupt 1 (CTRL_REG3) */
-    LIS2DH12_INT_TYPE_I1_OVERRUN = 0x02, /**< FIFO overrun interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_WTM     = 0x04, /**< FIFO watermark inter. on INT1 */
-    LIS2DH12_INT_TYPE_I1_ZYXDA   = 0x10, /**< ZYXDA interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_IA2     = 0x20, /**< IA2 interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_IA1     = 0x40, /**< IA1 interrupt on INT1 */
-    LIS2DH12_INT_TYPE_I1_CLICK   = 0x80, /**< click interrupt on INT1 */
-
-    /* for interrupt 2 (CTRL_REG6) */
-    LIS2DH12_INT_TYPE_INT_POLARITY = 0x02, /**< INT1 and INT2 pin polarity */
-    LIS2DH12_INT_TYPE_I2_ACT       = 0x08, /**< enable activity interrupt on INT2 */
-    LIS2DH12_INT_TYPE_I2_BOOT      = 0x10, /**< enable boot on INT2 */
-    LIS2DH12_INT_TYPE_I2_IA2       = 0x20, /**< IA2 on INT2 */
-    LIS2DH12_INT_TYPE_I2_IA1       = 0x40, /**< IA1 on INT2 */
-    LIS2DH12_INT_TYPE_I2_CLICK     = 0x80, /**< click interrupt on INT2 */
-};
-
 /**
  * @brief    Parameter for interrupt configuration
  */
 typedef struct {
     uint8_t int_config;             /**< values for configuration */
```

Comment:
Should this be `LIS2DH12_INT1_CFG_t`?

Reply:
Yes. And I think the `LIS2DH12_INT1_CFG_t` and `LIS2DH12_INT2_CFG_t` can be combined to `LIS2DH12_INT_CFG_t` because both registers have the same entries. This are only physically seperated registers to control each interrupt.